src/stories/utils/show-more.js

```diff
@@ -1,41 +1,62 @@
 document.addEventListener("DOMContentLoaded", () => {
-  const lists = document.querySelectorAll("[data-show-more-list]");
+  const listWrappers = document.querySelectorAll(
+    "[data-show-more-list-wrapper]"
+  );
 
-  lists.forEach((list) => {
-    const uid = Math.floor(Math.random() * 1000000);
-    list.setAttribute("id", `category-list-${uid}`);
+  listWrappers.forEach((listWrapper) => {
+    const list = listWrapper.querySelector("[data-show-more-list]");
+
+    const listElements = list.querySelectorAll("[data-show-more-item]");
+    const amountOfListElements = parseInt(listElements.length, 10);
+    const listShowMoreButton = listWrapper.querySelector(
+      "[data-show-more-button]"
+    );
 
-    const categories = list.querySelectorAll("[data-show-more-item]");
-    const toggleButton = list.querySelector("[data-show-more-button]");
-    toggleButton.setAttribute("id", `category-toggle-${uid}`);
+    if (!list || !listElements || !listShowMoreButton) return;
```

Comment:
Consider printing an error meessage to `drupal.debug()` if these expected elements are missing. This might make debugging easier instead of just "failing" silently.